File: Web/react-spa/src/features/products/product.ts
export default class Product {
    constructor(
        public id: string = '', 
        public name: string = '',
        public description: string = '',
        public price: number = 0,
        public picturesUris: Array<string> = [],
        public availableStock: number = 0,
        public brand: string  = '',
        public type: string  = '',
        public isArchived: boolean = false,
        public storeId: string = ''
    ) { }
}

File: Services/Catalog/Catalog.DAL/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Catalog.DAL.Entities
{
    public class Product : Entity
    {
        public string Name { get; set; }
        public string? Description { get; set; }
        public decimal Price { get; set; }
        public string[] PicturesUris { get; set; }
        public int AvailableStock { get; set; }
        public bool IsArchived { get; set; }


        [ForeignKey("Brand")]
        public Guid BrandId { get; set; }
        public Brand Brand { get; set; }

        [ForeignKey("Type")]
        public Guid TypeId { get; set; }
        public Type Type { get; set; }

        public Guid StoreId { get; set; }
    }
}
